backend/src/nodes/nodes/image_filter/blur_lens.py

```diff
@@ -0,0 +1,150 @@
+                total += p['A'] * (k[0,i].real*k[0,j].real - k[0,i].imag*k[0,j].imag) + p['B'] * (k[0,i].real*k[0,j].imag + k[0,i].imag*k[0,j].real)
+    scalar = 1 / math.sqrt(total)
+    kernels = np.asarray(kernels) * scalar
+    return kernels
+
+def weighted_sum(kernel, params):
+    return np.add(kernel.real * params['A'], kernel.imag * params['B'])
+
+def multiply_kernel(kernel):
+    kernel_size = kernel.shape[1]
+    a = np.repeat(kernel, kernel_size, 0)
+    b = np.repeat(kernel.transpose(), kernel_size, 1)
+    return np.multiply(a,b)
+
+def lens_blur(img, radius=3, components=5, exposure_gamma=5):
+    img = np.ascontiguousarray(img.transpose(2,0,1), dtype=np.float32)
+    parameters, scale = get_parameters(component_count = components)
+    components = [complex_kernel_1d(radius, scale, component_params['a'], component_params['b']) for component_params in parameters]
+    components = normalise_kernels(components, parameters)
+    img = np.power(img, exposure_gamma)
+    component_output = list()
+    for component, component_params in zip(components, parameters):
+        channels = list()
+        for channel in range(img.shape[0]):
+            inter = signal.convolve2d(img[channel], component, boundary='symm', mode='same')
+            channels.append(signal.convolve2d(inter, component.transpose(), boundary='symm', mode='same'))
+        component_image = np.stack([weighted_sum(channel, component_params) for channel in channels])
+        component_output.append(component_image)
+    output_image = reduce(np.add, component_output)
+    output_image = np.clip(output_image, 0, None)
+    output_image = np.power(output_image, 1.0/exposure_gamma)
+    output_image = np.clip(output_image, 0, 1)
+    output_image = output_image.transpose(1,2,0)
+    return output_image
+
+@NodeFactory.register("chainner:image:lens_blur")
+class LensBlurNode(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = "Apply Lens blur to an image."
+        self.inputs = [
+            ImageInput(),
+            NumberInput(
+                "Radius",
+                minimum=1,
+                default=3,
+                precision=1,
+                controls_step=1,
+            ),
+            SliderInput(
+                "Components",
+                minimum=1,
+                maximum=6,
+                default=5,
+                precision=0,
+                controls_step=1,
+            ),
+            NumberInput(
+                "Exposure Gamma",
+                minimum=0.01,
+                maximum=100,
+                default=5,
+                precision=4,
+                controls_step=0.1,
+            ),
+        ]
+        self.outputs = [ImageOutput(image_type="Input0")]
+        self.category = ImageFilterCategory
+        self.name = "Lens Blur"
+        self.icon = "MdBlurOn"
+        self.sub = "Blur"
+
+    def run(
+        self,
+        img: np.ndarray,
+        in_radius: int,
```

Comment:
This is a float because of `precision=1`.

```suggestion
        in_radius: float,
```

Reply:
Radius is int so removed precision